### PR TITLE
Implemented POM and fixture tests for e-commerce site

### PR DESCRIPTION
**Pull request by VasylBurych**
Screenshot_2

### CODE REVIEW

File: fixtures/loginFixture.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { test as base } from '@playwright/test';
+import dotenv from '.env.example';
+
+dotenv.config();
+
+const username = process.env.USERNAME;
+const password = process.env.PASSWORD;
+
+export const test = base.extend({
+    loginPage: async ({ page }, use) => {
```
Comment (ybaranovskyi11) [inline fixtures/loginFixture.ts]: login method should be in POM page, and here in fixtures you import this page, and use it

File: page-objects/HomePage.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+export class HomePage {
+    readonly page: Page;
+
+    constructor(page: Page) {
+        this.page = page;
+    }
+
+    async goToWomenCategory() {
+        await this.page.click('text=Women');
```
Comment (ybaranovskyi11) [inline page-objects/HomePage.ts]: locator inside params is a bad practice, update it

File: page-objects/HomePage.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        //await this.page.click('text=Hoodies & Sweatshirts');
```
Comment (ybaranovskyi11) [inline page-objects/HomePage.ts]: commented code

File: page-objects/LoginPage.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+export class LoginPage {
+    readonly page: Page;
+
+    constructor(page: Page) {
+        this.page = page;
+    }
+
+    async login(username: string, password: string) {
+        await this.page.goto('https://magento.softwaretestingboard.com/');
+        await this.page.click('text=Sign In');
```
Comment (ybaranovskyi11) [inline page-objects/LoginPage.ts]: here too, dont use locator inside params

File: page-objects/ProductPage.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+export class ProductPage {
+    readonly page: Page;
+
+    constructor(page: Page) {
+        this.page = page;
+    }
+
+    async addToWishlist() {
+        await this.page.waitForSelector('a[title="Add to Wish List"]', { state: 'visible' });
+        await this.page.click('a[title="Add to Wish List"]', { force: true });
```
Comment (ybaranovskyi11) [inline page-objects/ProductPage.ts]: locator inside params

File: page-objects/WishlistPage.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+export class WishlistPage {
+    readonly page: Page;
+
+    constructor(page: Page) {
+        this.page = page;
+    }
+
+    async openWishlist() {
+        await this.page.click('text=My Wishlist');
```
Comment (ybaranovskyi11) [inline page-objects/WishlistPage.ts]: here too, locator inside params

File: tests/e-commerce/e-commerce.spec.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import { test } from './fixtures';
+import { HomePage } from '../../page-objects/HomePage';
+import { ProductPage } from '../../page-objects/ProductPage';
+import { WishlistPage } from '../../page-objects/WishlistPage';
+
+test.describe('E-commerce Test', () => {
+    test('Add product to Wishlist', async ({ loggedInPage }) => {
+        const homePage = new HomePage(loggedInPage);
```
Comment (ybaranovskyi11) [inline tests/e-commerce/e-commerce.spec.ts]: its not a test fixtures, update it

File: tests/e-commerce/fixtures.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import { Page, test as baseTest } from '@playwright/test';
```
Comment (ybaranovskyi11) [inline tests/e-commerce/fixtures.ts]: in this case, it should be one fixtures file

**Comment by VasylBurych** [comment]
Fixed
image

**Pull request merged**